nri-postgresql/src/Postgres/Enum.hs

```diff
@@ -0,0 +1,242 @@
+{-# LANGUAGE TemplateHaskell #-}
+{-# OPTIONS -Wno-incomplete-uni-patterns #-}
+{-# OPTIONS -Wno-orphans #-}
+
+module Postgres.Enum where
+
+import qualified Environment
+import qualified Postgres.Settings
+import qualified Data.ByteString.Lazy as BSL
+import Data.List (group)
+import qualified Data.Text.Encoding as Encoding
+import qualified Language.Haskell.TH as TH
+import Database.PostgreSQL.Typed.Protocol (pgSimpleQuery)
+import Database.PostgreSQL.Typed.TH (withTPGConnection, useTPGDatabase)
+import Database.PostgreSQL.Typed.Dynamic (pgDecodeRep, PGRepType, PGRep)
+import Database.PostgreSQL.Typed.Types (PGType, PGVal, PGParameter, pgEncode, PGColumn, pgDecode)
+import qualified Set
+import qualified Prelude 
+
+quote :: Prelude.String -> Prelude.String
+quote t = "\"" ++ t ++ "\""
+
+makeValueList :: List Text -> Text
+makeValueList list = 
+  "[" ++ Text.join ", " (List.map (Text.toList >> quote >> Text.fromList) list) ++ "]"
```

Comment:
I think we don't need to go from Text to String and back, we can just change quote to take and return Text.

Reply:
Isn't this the same as `show`?

Reply:
like `makeConList = Prelude.show`?

Reply:
That's..... a great point lol. Thanks!